src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs

```diff
@@ -570,9 +572,31 @@ impl Persist for VirtSandbox {
         let sandbox_state = crate::sandbox_persist::SandboxState {
             sandbox_type: VIRTCONTAINER.to_string(),
             resource: Some(self.resource_manager.save().await?),
-            hypervisor: Some(self.hypervisor.save_state().await?),
+            //hypervisor: Some(self.hypervisor.save_state().await?),
+            hypervisor: match self.hypervisor.save_state().await?.hypervisor_type.as_str() {
+                // TODO support other hypervisors
+                HYPERVISOR_DRAGONBALL => Ok(Some(self.hypervisor.save_state().await?)),
+                HYPERVISOR_FIRECRACKER => Ok(Some(self.hypervisor.save_state().await?)),
```

Comment:
This looks odd to me, why is `save_state()` called again?  If the purpose of the first call (in the `match` expression above) is only to get the hypervisor type, isn't there a more direct way to do that?

Reply:
exactly! maybe we should call save state earlier and keep it as a variable to pass it on there. Not sure, what do you think?

Reply:
ACK, thanks!